X10D/src/Reflection/MemberInfoExtensions.cs

```diff
@@ -82,7 +82,7 @@ public static bool HasCustomAttribute(this MemberInfo member, Type attribute)
                 attribute, typeof(Attribute)), nameof(attribute));
         }
 
-        return member.GetCustomAttribute(attribute) is not null;
+        return Attribute.IsDefined(member, attribute);
```

Comment:
Is there any particular reason for this change? This also applies to the change on line 41.

Reply:
Yes, if member has our needed attribute type, the runtime will create a new instance of that attribute. Since we have no usage of it beside null-checking, imo, checking IsDefined is better.